docs/ionq/jobs.md

```diff
@@ -0,0 +1,120 @@
+on the scheduler, the job is assigned an id and then one can query this
+job via the API. The job has a status on it which tells what state the job is in
+`running`, `completed`, `failed`, etc.  From a users perspective this is abstracted
+mostly away in Cirq, depending on whether one wants to make blocking calls (which
+wait for the job to complete), or non-blocking calls (where one creates a job and
+only later does one query for the results.)
+
+Here we describe these different methods.
+
+### Via Run
+
+The first method for running is to do so via the `run` method on `cirq.ionq.Service`.
+
+```python
+qubit = cirq.LineQubit(0)
+circuit = cirq.Circuit(
+    cirq.X(qubit)**0.5,            # Square root of NOT.
+    cirq.measure(qubit, key='x')   # Measurement store in key 'x'
+)
+
+result = service.run(circuit=circuit, repetitions=100, target='qpu')
+print(result)
+```
+Which results in
+```
+x=0000000000000000000000000000000000000000000000000000111111111111111111111111111111111111111111111111
+```
+Looking at these results you should notice something strange. What are the odds
+that the x measurements were all 0s followed by all 1s?  The reason for this
+sorting is that the IonQAPI only returns statistics about the results, i.e. what
+count of results were 0 and what count were 1 (or if you are measuring
+multiple qubits the counts of the different outcome bit string outcomes).  In
+order to make this compatible with Cirq's notition of `cirq.Result`, these
+are then converted into raw results with the exactly correct number of
+results (in lexical order). In other words, the measurement results are not
+in an order corresponding to the temporal order of the measurements.
+
+When calling run, one will need to include the number of `repetitions` or shots
+for the given circuit.  In addition, if there is no `default_target` set on the
+service, then a `target` needs to be specified.  Currently the supported targets
+are `qpu` and `simulator`.
+
+### Via a sampler
+
+Another method to get results from the IonQ API is to use a sampler.  A sampler
+is specifically design to be a lightweight interface for obtaining results
+in a [pandas](https://pandas.pydata.org/) dataframe and is the interface
+used by other classes in Cirq for objects that process data.  Here is a
+simple example showing how to get a sampler and use it.
+
+```python
+qubit = cirq.LineQubit(0)
+circuit = cirq.Circuit(
+    cirq.X(qubit)**0.5,            # Square root of NOT.
+    cirq.measure(qubit, key='x')   # Measurement store in key 'x'
+)
+sampler = service.sampler(target='qpu')
+result = sampler.run(program=circuit, repetitions=100)
+print(result)
+```
+
+### Via create job
+
+The above two methods, using run and the sampler, both block waiting for
+results.  This can be problematic when the queueing time for the service
+is long.  Instead it is recommended that one use the job api directly.
+In this pattern one first create the job with the quantum circuit one
```

Comment:
`one first ` -> `you can first`

Reply:
Doug has now asked for more comma's than my thesis advisor.